### PR TITLE
Support Ecto v3.4.2 and rewrite ExAudit.Repo to extend Ecto.Repo

### PR DESCRIPTION
**Pull request by hisapy**
This is a breaking change to support ecto and ecto_sql ~> 3.4 and at the
same time to actually use ExAudit.Repo with Ecto.Repo.

The Ecto.Repo @callback implementations are overriden to add version tracking if the
given struct or changeset is in the list of :tracked_schemas specified in the config
or call the original Ecto.Repo implementation otherwise.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,36 +6,36 @@ ExAudit plugs right into your ecto repositories and hooks all the data mutating
...
 ## Features
- 
- * Wraps Ecto.Repo, no need to change your existing codebase to start tracking changes
- * Creates +- diffs of the casted structs. Custom types are automatically supported.
- * Ships with functions to review the history of a struct and roll back changes
- * Allows custom ID types and custom fields in the version schema
- * Tracks associated entities when they're created, updated or deleted in a single Repo call
- * Recursively tracks cascading deletions
+
+- Wraps Ecto.Repo, no need to change your existing codebase to start tracking changes
```
Comment (hisapy) [inline README.md:10]: My IDE did this changes (- instead of *) automatically when I saved the file. I hope you don't mind 😅

Comment (narrowtux) [reply README.md:10]: All this reformating makes it really hard to see what you actually changed.

**Comment by narrowtux** [comment]
I like what you did there, especially that we don't need to reimplement all repo functions any more.

File: README.md (hunk 1 above)
Comment (hisapy) [reply README.md:10]: Sorry, I don't know why my VS Code formatter changed those.

**Comment by hisapy** [comment]
Yeah, I upgraded Ecto and started using Oban and got the debug log from the queues outputted in the console when they shouldn't so [I replied a similar issue](https://github.com/sorentwo/oban/issues/23#issuecomment-614194730) and the maintainer told me that it might be something with prepare_query.

So after digging around I realized that it'd be better to avoid rewriting the whole Ecto.Repo @callback implementations to make ExAudit more easy to maintain with respect to Ecto upgrades and after a couple of days and designs I came up with this solution. 

Thx for this lib. I like it because of the clever way of auditing transparently and supporting assocs using the children traversal logic that is already implemented in Ecto.Repo.Schema.

**Comment by yukster** [comment]
@narrowtux any chance we could get a new release? I have a working integration based off master (using Ecto 3.4.2) but I'd rather not leave it that way if I can change to a tag.

**Comment by yukster** [comment]
Duh, I can just use the :ref option. Still, working with 3.4 is a big deal and seems worth a release.

BTW, nice work on this lib! It's slick.

**Comment by cleveyeo** [comment]
Is there going to be a new release? am in need of this to work with Ecto 3.4

**Pull request merged**